Store sessions info separately per tracker namespace (close #570)

AlexBenny opened the pull request:
Multiple tracker instances needs to have independent session info.
The tracker 1.x used to store some information about the session in the filesystem.
That would cause interferences between trackers in a multi-tracker app.

paulboocock reviewed (commented):
Looks like your new test is failing on iOS12? Not immediately obvious why however.

AlexBenny commented:
> Looks like your new test is failing on iOS12? Not immediately obvious why however.

It was failing because of a bug not caught by the old tests. The new test I've added made it visible.
I think it only affects the Simulator but I've added a fix in the session management.
Also I changed the place where we store the session info from NSDocumentDirectory to NSApplicationSupportDirectory.
Which is better because the document directory is accessible directly by the user.
